core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -318,14 +318,20 @@ private ServiceInfo addServiceExamples(ServiceInfo service) {
                 service.methods().stream().map(m -> new MethodInfo(
                         m.name(), m.returnTypeSignature(), m.parameters(), m.exceptionTypeSignatures(),
                         m.endpoints(),
-                        Iterables.concat(m.exampleHttpHeaders(), exampleHttpHeaders.get(m.name())),
-                        Iterables.concat(m.exampleRequests(), exampleRequests.get(m.name())),
+                        // Show the examples added via `DocServiceBuilder` before the examples
+                        // generated by the plugin.
+                        concatAndDedup(exampleHttpHeaders.get(m.name()), m.exampleHttpHeaders()),
+                        concatAndDedup(exampleRequests.get(m.name()), m.exampleRequests()),
                         m.httpMethod(), m.docString()))::iterator,
                 Iterables.concat(service.exampleHttpHeaders(),
                                  exampleHttpHeaders.get("")),
                 service.docString());
     }
 
+    private static <T> Iterable<T> concatAndDedup(Iterable<T> first, Iterable<T> second) {
+        return Streams.stream(Iterables.concat(first, second)).distinct()::iterator;
```

Comment:
A bit more characters, but guess we may just stick to Stream API for simplicity `Stream.concat(Streams.stream(first), Streams.stream(second)).distinct()`